src/managers/GitObjectManager.js

```diff
@@ -77,7 +71,20 @@ export class GitObjectManager {
     let { type, object } = GitObject.unwrap({ oid, buffer })
     if (format === 'content') return { type, format: 'content', object, source }
   }
-
+  static async loadPack (fs, gitdir, filename, getExternalRefDelta) {
+    // If not there, load it from a .idx file
+    const idxName = filename.replace(/pack$/, 'idx')
+    if (await fs.exists(`${gitdir}/objects/pack/${idxName}`)) {
+      const idx = await fs.read(`${gitdir}/objects/pack/${idxName}`)
+      return GitPackIndex.fromIdx({ idx, getExternalRefDelta })
+    }
+    // If the .idx file isn't available, generate one.
+    const pack = await fs.read(`${gitdir}/objects/pack/${filename}`)
+    const p = GitPackIndex.fromPack({ pack, getExternalRefDelta })
+    // Save .idx file
+    fs.write(`${gitdir}/objects/pack/${idxName}`, (await p).toBuffer())
```

Comment:
Why is the `await` on this line instead of on the previous line, `const p = GitPackIndex.fromPack(`?